### PR TITLE
feat(api): add monthly closing checklist tool

### LINKED ISSUE #99: feat: Add monthly closing checklist tool (月次決算チェックリスト)
## Summary

複数のfreee APIを組み合わせた統合チェックリストツールを追加し、月次決算の網羅的な確認を1コマンドで実行可能にする。

### 税理士・経営者の課題
- 月次決算は毎月繰り返す定型業務だが、確認項目が多く漏れが発生しやすい
- 主な確認事項：
  - 未処理の銀行明細がないか
  - 前月残高と当月期首残高が一致するか
  - 仮勘定（仮払金・仮受金等）の残高が適切か
  - 売掛金・買掛金の滞留がないか
  - 未紐付けの証憑がないか
- 現状、これらを個別にAPIを呼んで確認する必要があり、手間がかかる
- 1つのツールで網羅的にチェックし、問題箇所をハイライトできれば大幅に効率化

### 活用するfreee API
- `GET /api/1/trial_balance` — 残高試算表（各勘定科目の残高確認）
- `GET /api/1/reports/trial_bs` — 貸借対照表
- `GET /api/1/deals` — 取引一覧（未消込取引の検出）
- `GET /api/1/wallet_txns` — 明細一覧（未処理明細の検出）
- `GET /api/1/receipts` — 証憑（未紐付けチェック）※ #98 実装後

### New Tool
- `freee_monthly_closing_check` — 月次決算チェックリスト実行

### Input Schema
```typescript
{
  companyId?: number;
  year: number;
  month: number;        // 1-12
  checks?: Array<       // 実行するチェック項目を指定（省略時は全チェック）
    | 'unprocessed_transactions'  // 未処理明細チェック
    | 'balance_verification'      // 残高突合チェック
    | 'temporary_accounts'        // 仮勘定残高チェック
    | 'receivable_aging'          // 売掛金滞留チェック
    | 'payable_aging'             // 買掛金滞留チェック
    | 'unattached_receipts'       // 未紐付け証憑チェック
  >;
}
```

### Output Format
LLMが分析・報告しやすい構造化された結果を返す：

```typescript
{
  period: "2024年1月",
  overallStatus: "warning",  // ok | warning | error
  checks: [
    {
      name: "未処理明細チェック",
      status: "warning",
      details: "3件の未処理明細があります",
      items: [...]
    },
    {
      name: "残高突合チェック",
      status: "ok",
      details: "全口座の残高が一致しています"
    },
    // ...
  ],
  summary: "6項目中5項目OK、1項目で要対応があります"
}
```

### チェックロジック

1. **未処理明細チェック**: `wallet_txns` でステータスが未処理のものを検出
2. **残高突合チェック**: `trial_balance` の現金預金科目残高と `walletables` の口座残高を比較
3. **仮勘定残高チェック**: `trial_balance` で仮払金・仮受金・立替金の残高を確認（0でなければ警告）
4. **売掛金滞留チェック**: `deals` で type=income かつ未消込の取引を日数別に集計
5. **買掛金滞留チェック**: `deals` で type=expense かつ未消込の取引を日数別に集計
6. **未紐付け証憑チェック**: `receipts` で unconfirmed の件数を確認

### API呼び出し最適化
- チェック項目ごとに必要なAPIのみ呼び出す
- 並列実行可能なAPIは `Promise.all` で並列化
- 合計API呼び出し数を最小限に抑える（6項目全実行でも10回以内を目標）

### 1. 月次決算の完了確認
```
User: "1月の月次決算チェックをして"
→ freee_monthly_closing_check で全項目チェック
→ 結果をサマリー表示：
  ✅ 未処理明細: なし
  ✅ 残高突合: 全口座一致
  ⚠️ 仮勘定: 仮払金 50,000円の残高あり
  ✅ 売掛金: 滞留なし
  ✅ 買掛金: 滞留なし
  ✅ 証憑: 全件紐付け済み
```

### 2. 特定項目のみチェック
```
User: "売掛金の滞留状況だけ確認して"
→ freee_monthly_closing_check で receivable_aging のみ実行
```

## Acceptance Criteria
- [ ] `freee_monthly_closing_check` で月次決算チェックを実行できる
- [ ] 6種類のチェック項目が動作する
- [ ] チェック項目の選択指定が可能
- [ ] 結果がLLMフレンドリーな構造化データで返される
- [ ] API呼び出し数が最適化されている
- [ ] Zod スキーマによるバリデーション
- [ ] テスト追加
- [ ] `npm run build && npm run typecheck && npm run lint && npm test` がパスする

## Labels
enhancement

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

Implements `freee_monthly_closing_check` MCP tool (Issue #99) that runs automated month-end closing checks against freee APIs.

- **6 check types**: unprocessed transactions, balance verification, temporary accounts (仮払金/仮受金/立替金), receivable aging, payable aging, unattached receipts (stub pending #98)
- **Parallel API calls**: Only fetches data needed for selected checks, minimizing API usage
- **Structured results**: Each check returns ok/warning/error status with Japanese details and actionable items
- **Selective execution**: Optional `checks` parameter to run specific checks only

### Files Changed (6)
| File | Change |
|------|--------|
| `src/types/freee.ts` | Add `MonthlyClosingCheckType`, `MonthlyClosingCheckItem`, `MonthlyClosingCheckResult` types |
| `src/schemas.ts` | Add `MonthlyClosingCheckSchema` with year/month/checks params |
| `src/api/freeeClient.ts` | Add `getMonthlyClosingChecklist()` + 6 private check methods |
| `src/index.ts` | Register `freee_monthly_closing_check` as tool #40 |
| `src/__tests__/api/monthlyClosingCheck.test.ts` | 14 unit tests covering all check types |
| `src/__tests__/index.test.ts` | Update tool/schema count assertions (39→40) |

## Test plan

- [x] 14 new tests for monthly closing check (all checks, edge cases, date handling)
- [x] 334 total tests passing (0 failures)
- [x] TypeScript compiles cleanly
- [x] ESLint: 0 errors (only pre-existing warnings)
- [x] Pre-commit hooks pass (eslint, typecheck, gitleaks)

Closes #99

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/api/freeeClient.ts [hunk 1]
```diff
@@ -1538,5 +1541,373 @@ export class FreeeClient {
...
+  private checkAgingDeals(
+    deals: FreeeDeal[],
+    type: 'income' | 'expense',
+    referenceDate: string,
+  ): MonthlyClosingCheckItem {
+    const name =
+      type === 'income' ? '売掛金滞留チェック' : '買掛金滞留チェック';
+    const unsettled = deals.filter(
+      (d) => d.type === type && d.status !== 'settled',
+    );
```
[the remark this answers is not in the document]
Comment (knishioka) [reply src/api/freeeClient.ts]: Fixed in commit 6a6283e. Added `d.issue_date <= referenceDate` filter to exclude deals issued after the reference date, preventing negative days_outstanding and inflated aging totals for backdated closing runs.
